packages/fuel-gauge/src/predicate.test.ts

```diff
@@ -352,4 +441,56 @@ describe('Predicate', () => {
 
     expect(failed).toEqual(true);
   });
+
+  it('can call a Coin predicate which returns true with valid predicate data [main args struct]', async () => {
+    const receiverAddress = Address.fromRandom();
+    const wallet = await setup();
+    const amountToPredicate = 10;
+    const predicate = new Predicate(testPredicateStruct, predicateMainArgsStructAbi);
+
+    const initialPredicateBalance = await setupPredicate(wallet, amountToPredicate, predicate);
+    const initialReceiverBalance = await wallet.provider.getBalance(receiverAddress, NativeAssetId);
+
+    const validation: Validation = {
+      has_account: true,
+      total_complete: 100,
+    };
+    await wallet.provider.submitSpendPredicate(
```

Comment:
We should add a type optional argument here to inform the Validation types. It will help to generate better types after.